### ISSUE TITLE
Mock NgRedux does not compile in strict mode

### ISSUE THREAD

**Issue by ialibhay** [angular-redux/store#415]
### This is a...

* [ ] feature request
* [x] bug report
* [ ] usage question

### What toolchain are you using for transpilation/bundling?

* [x] `@angular/cli`
* [ ] Custom `@ngTools/webpack`
* [ ] Raw `ngc`
* [ ] SystemJS
* [ ] Rollup
* [ ] Other

### Environment

NodeJS Version: 6.9.5
Typescript Version: 2.3.3
Angular Version: 4.1.3
@angular-redux/store version: 6.4.2
@angular/cli version: 1.1.0
OS: macOS 10.12.5

### Expected Behaviour:
Expect no compile errors when importing from `@angular-redux/store/testing` when compiling TypeScript with the `strict` flag.

### Actual Behaviour:
Compile errors are showing when setting the `strict` flag.

### Stack Trace/Error Message:
```
ERROR in /node_modules/@angular-redux/store/testing/observable-store.mock.ts (2,3): 'NgRedux' is declared but never used.

ERROR in /node_modules/@angular-redux/store/testing/observable-store.mock.ts (10,19): 'Action' is declared but never used.

ERROR in /node_modules/@angular-redux/store/testing/observable-store.mock.ts (47,14): Function expression, which lacks return-type annotation, implicitly has an 'any' return type.

ERROR in /node_modules/@angular-redux/store/testing/observable-store.mock.ts (47,15): Parameter 'action' implicitly has an 'any' type.

ERROR in /node_modules/@angular-redux/store/testing/observable-store.mock.ts (48,20): Function expression, which lacks return-type annotation, implicitly has an 'any' return type.

ERROR in /node_modules/@angular-redux/store/testing/observable-store.mock.ts (49,14): Function expression, which lacks return-type annotation, implicitly has an 'any' return type.

ERROR in /node_modules/@angular-redux/store/testing/observable-store.mock.ts (50,21): Function expression, which lacks return-type annotation, implicitly has an 'any' return type.

ERROR in /node_modules/@angular-redux/store/testing/ng-redux.mock.ts (5,3): 'ObservableStore' is declared but never used.

ERROR in /node_modules/@angular-redux/store/testing/ng-redux.mock.ts (9,10): 'Reducer' is declared but never used.

ERROR in /node_modules/@angular-redux/store/testing/ng-redux.mock.ts (9,19): 'Action' is declared but never used.

ERROR in /node_modules/@angular-redux/store/testing/ng-redux.mock.ts (10,10): 'Observable' is declared but never used.

ERROR in /node_modules/@angular-redux/store/testing/ng-redux.mock.ts (12,10): 'ReplaySubject' is declared but never used.
```

### Additional Notes:
This only seems to happen for the testing module, as the import resolves the source files directly instead of the distributables under `lib/testing`. The standard module is resolved appropriately.

**Comment by SethDavenport** [comment]
This ties in to a broader effort to turn on `strict` in our build process. Working on it now.

**Comment by ialibhay** [comment]
This is also a packaging issue, if the .d.ts files were being resolved for the testing module, this likely wouldn't occur.

I don't get this issue at all with the primary module in the actual app.

**Comment by SethDavenport** [comment]
The `index.d.ts` file for the main module code is generated from the exports in `src/index.ts`.  This is set up as the `typings` file in `package.json`.

I don't want to include the `testing` stuff in there because users should not have to import the test files at runtime.

I could generate and package `.d.ts` files for the testing module, but it's not clear consumers will find them if they're not referenced in `package.json`.  Unless I'm missing something.

**Comment by ialibhay** [comment]
I was looking at how Angular itself packages its testing bundle, it seems that there is a `testing.d.ts` exposed in the root folder along with a `testing` folder which contains additional `.d.ts` files. This folder then has its own `package.json` which points to the testing bundle rather than the main one, so that imports from `@angular/{module}/testing` correctly import the testing bundle. See: https://github.com/angular/angular/blob/master/packages/core/testing/package.json

A similar packaging strategy here would make it easier to support library users who use any kind compiler options.

However, I'm not convinced that the effort of setting up this fancier packaging is worth the separate effort in addition to your current work in enabling the strict flag in the library. Maybe as a nice-to-have in the future for users who don't use TypeScript with angular-redux?

**Comment by ialibhay** [comment]
Actually, I just did a quick test by adding a `package.json` to the testing folder with the following content:

```
{
  "name": "@angular-redux/store/testing",
  "typings": "../lib/testing/index.d.ts",
  "main": "../lib/testing/index.js"
}
```

and the tests now compile in strict mode.

**Comment by SethDavenport** [comment]
Cool. Mind PR-ing it up?

**Comment by SethDavenport** [comment]
Fix release in 6.4.4
